src/Sylius/Bundle/AddressingBundle/Form/EventListener/BuildAddressFormSubscriber.php

```diff
@@ -96,13 +99,17 @@ public function preSubmit(FormEvent $event)
             return;
         }
 
-        $country = $this->countryRepository->find($data['country']);
+        if ('' === $data['country']) {
+            return;
+        }
+
+        $country = $this->countryRepository->findOneBy(array('code' => $data['country']));
         if (null === $country) {
             return;
         }
 
         if ($country->hasProvinces()) {
-            $event->getForm()->add($this->factory->createNamed('province', 'sylius_province_choice', null, array(
+            $event->getForm()->add($this->formFactory->createNamed('province', 'sylius_province_code_choice', null, array(
```

Comment:
I think this is duplicated with preSetData, can we extract it to private method?